plugins.py

```diff
@@ -0,0 +1,82 @@
+import time
+import datetime
+import deltachat
+
+
+class ReceivePlugin:
+    name = "test account"
+
+    @staticmethod
+    @deltachat.account_hookimpl
+    def ac_incoming_message(message):
+        received = time.time()
+        message.create_chat()
+        if not message.chat.can_send():
+            # if it's a device message or mailing list, we don't need to look at it.
+            return
+        sender = message.get_sender_contact().addr
+        receiver = message.account.get_self_contact().addr
+        msgcontent = parse_msg(message.text)
+        firsttravel = msgcontent.get("testduration")
+        secondtravel = received - msgcontent.get("begin")
+        iam = msgcontent.get("sender")
+        if message.chat.is_group():
+            if iam == "spider":
+                print("%s: joined group chat %s after %.1f seconds" % (receiver, message.chat.get_name(), secondtravel))
+                message.chat.send_text("Sender: %s\nBegin: %s" % (receiver, str(time.time())))
+                return  # for now it's enough to just accept the group chat
+            else:
+                return  # when messages by others arrived is checked later by the main thread
+        print("%s: test message took %.1f seconds to %s and %.1f seconds back." %
+              (receiver, firsttravel, sender, secondtravel))
```

Comment:
It would be nice if we could print the message size or attachment size here - to explain why the message took so long ;)